Automattic/simplenote-macos
# New Auth Flow

### Fix
In this PR we're (heavily) refactoring the Authentication UI:

1. **LoginWindowController** has been split into **AuthWindowController** / **AuthViewController**
2. **Most of the code in AuthViewController hasn't been changed**
3. Autolayout has been implemented (YAY!)
4. Clicking the `SignUp` button will now hide the **Password** + **Forgot** + **Dotcom SSO** elements

Apologies about the magnitude of the PR, I really did try to split things up, buuut I couldn't figure a way that made sense!
**Most of the code** has been unchanged (moved from LoginWindowController).

cc @eshurakov 



## Testing

### IMPORTANT
This PR is **just UI**. The SignUp flow isn't functional at this stage!

### Test: SignUp UI
0. Ensure you're logged out!
1. Click over the **Sign Up** button

- [x] Verify that the **Password** + **Forgot Password** + **LogIn with WordPress.com** fields are hidden
- [x] Verify the Action Button switches to **Sign Up**
- [x] Verify that the legend at the bottom of the UI reads as **Already have an account? Log In**

### Test: Auth Repositioning
0. Ensure you're logged out!
1. Launch Simplenote
2. Verify the Auth UI is centered onscreen
3. Move the Auth UI anywhere else
4. Enter invalid credentials

- [x] Verify an error shows up on top
- [x] Verify the Auth UI doesn't get repositioned

### Test: Auth
0. Ensure you're logged out!
1. Launch Simplenote
2. Enter your Username
3. Enter your password
4. Press **Return**

- [x] Verify that you end up logged in!

### Test: Forgot Password
0. Ensure you're logged out!
1. Enter your email
2. Click over the **Forgot Password** field

- [x] Verify that the app redirects you to the Forgot Password Web, **with your email prefilled**

### Test: WordPress SSO
0. Ensure you're logged out!
1. Click over the **Log in with WordPress.com**
2. Follow the Dotcom LogIn flow

- [x] Verify you can log into your account!

### Release
These changes do not require release notes.

## Interface

### Screenshots
Old LogIn|New LogIn
-|-
<img width="300" alt="OldLogIn" src="https://user-images.githubusercontent.com/1195260/109842195-96794c80-7c28-11eb-93bc-1b5f154e4362.png">|<img width="300" alt="NewLogIn" src="https://user-images.githubusercontent.com/1195260/109842215-9bd69700-7c28-11eb-9f43-c4e86a73e818.png">

Old SignUp|New SignUp
-|-
<img width="300" alt="OldSignUp" src="https://user-images.githubusercontent.com/1195260/109842270-a8f38600-7c28-11eb-8d9e-c9454ad6f570.png">|<img width="300" alt="NewSignUp" src="https://user-images.githubusercontent.com/1195260/109842280-ab55e000-7c28-11eb-9787-c6c5f75194bc.png">
